### PR TITLE
perf: vectorize similarity calculations in check_duplicate using tensor batching #2655

### LINKED ISSUE #2655: perf: vectorize cos_sim calculation using tensor batching in check_duplicate
### Description
In `DuplicateService.check_duplicate()`, the application currently uses a Python `for` loop to sequentially calculate cosine similarity against every single stored ticket execution block:

```python
# CURRENT LINEAR LOOP DETECTED
best_score = 0.0
best_id = None

for ticket_id, stored_emb, _ in self._tickets:
    score = util.cos_sim(query_embedding, stored_emb).item()
    if score > best_score:
        best_score = score
        best_id = ticket_id
```
Python loops defeat the purpose of utilizing PyTorch and vectorization. Running individual scalar computations causes a linear execution slowdown processing latency) as the in-memory cache grows, which will degrade performance under high-throughput request loads.

Proposed Solution
Refactor Internal Data Structures: Instead of a list of tuples, maintain parallel arrays for structural components (list[str] for IDs) and stack vectors into a multi-dimensional matrix tensor (torch.Tensor).

Dynamic Matrix Appending: When calling add_ticket(), use torch.cat() to append the new tensor array row vertically along dim=0.

Vectorized Matrix Multiplication: In check_duplicate(), pass the query_embedding and the consolidated embeddings_tensor to util.cos_sim() once. Extract the highest matching vector score index instantly using torch.argmax().

### PR DESCRIPTION
**Pull request by suhaniiz**
## Description
This PR addresses a performance issue in `DuplicateService.check_duplicate()` where a linear Python `for` loop was utilized to evaluate cosine similarity sequentially. This manual iteration scaling pattern bypassed PyTorch tensor optimizations and increased processing latency as data volume grew.

## Changes
- **Parallel Tracker Redesign**: Changed tracking structures from a list of tuples to parallel groupings: a simple list tracking `_ticket_ids` and a multi-dimensional matrix tensor tracking `_embeddings_tensor`.
- **Dynamic Structural Stacking**: Adjusted historical loading workflows to utilize `torch.stack()` and new runtime items to vertically scale rows across `dim=0` using `torch.cat()`.
- **Vectorized Calculations**: Rewrote `check_duplicate()` to process similarity matrix math across the entire matrix stack instantly in a single operation, using `torch.argmax()` for immediate maximum index lookup.

## Related Issues
Closes #2655

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed email notification routing logic to correctly handle digest frequency mismatches in both directions.
  * Improved error handling for system settings retrieval with graceful fallbacks.

* **Performance & Reliability**
  * Enhanced duplicate detection with vectorized operations for improved efficiency and thread-safety mechanisms.
  * Added bounded caching for system settings to optimize repeated lookups.

### CODE REVIEW

**Comment by suhaniiz** [comment]
Hello @ritesh-1918 
This pull request has been raised as part of GSSoC 2026. Please review it when convenient and assign the appropriate labels.

**Comment by suhaniiz** [comment]
@ritesh-1918

**Comment by riteshbonthalakoti** [comment]
Hi @suhaniiz! Thanks for the amazing contribution. I've successfully merged your PR! 🚀

Please make sure to sign up under the company **Ritesh PVT Limited** when testing your features!

⚠️ Also, please complete the mandatory onboarding steps so your points register correctly:

⚠️ **MANDATORY GSSOC ONBOARDING STEPS**:
Before your PR points are finalized on the leaderboard, you **MUST** complete these steps:
1. ⭐ **Star this repository**: https://github.com/ritesh-1918/HELPDESK.AI (Mandatory)
2. 👤 **Follow the Project Admin**: https://github.com/ritesh-1918 (Mandatory - manual step)
3. 💼 **Connect on LinkedIn**: https://www.linkedin.com/in/ritesh1908/ (Mandatory)


Let's build something epic! 🔥
